File: ros2_ws/src/teleop_ik/test/test_target_msg.py
"""TargetPoseWithInput msg がビルド成果物から import できることを確認.

このテストは colcon build 完了後の環境でのみ pass する.
"""

from teleop_ik.msg import TargetPoseWithInput  # type: ignore[attr-defined]


def test_target_pose_with_input_instantiation():
    msg = TargetPoseWithInput()
    assert hasattr(msg, "header")
    assert hasattr(msg, "pose")
    assert hasattr(msg, "stick_x")
    assert hasattr(msg, "stick_y")
    assert msg.stick_x == 0.0
    assert msg.stick_y == 0.0


def test_target_pose_with_input_set_fields():
    msg = TargetPoseWithInput()
    msg.stick_x = 0.5
    msg.stick_y = -0.25
    assert msg.stick_x == 0.5
    assert msg.stick_y == -0.25

File: ros2_ws/src/teleop_ik/test/test_target_msg.cpp
// teleop_ik/test/test_target_msg.cpp
#include <gtest/gtest.h>

#include "teleop_ik/msg/target_pose_with_input.hpp"

TEST(TargetPoseWithInputMsg, DefaultValues)
{
  teleop_ik::msg::TargetPoseWithInput msg;
  EXPECT_TRUE(msg.stick_x == 0.0f);
  EXPECT_TRUE(msg.stick_y == 0.0f);
  EXPECT_EQ(msg.pose.position.x, 0.0);
  EXPECT_EQ(msg.pose.position.y, 0.0);
  EXPECT_EQ(msg.pose.position.z, 0.0);
}

TEST(TargetPoseWithInputMsg, SetFields)
{
  teleop_ik::msg::TargetPoseWithInput msg;
  msg.stick_x = 0.5f;
  msg.stick_y = -0.25f;
  msg.pose.position.x = 1.0;
  msg.pose.position.y = 2.0;
  msg.pose.position.z = 3.0;
  EXPECT_FLOAT_EQ(msg.stick_x, 0.5f);
  EXPECT_FLOAT_EQ(msg.stick_y, -0.25f);
  EXPECT_DOUBLE_EQ(msg.pose.position.x, 1.0);
  EXPECT_DOUBLE_EQ(msg.pose.position.y, 2.0);
  EXPECT_DOUBLE_EQ(msg.pose.position.z, 3.0);
}
